python/pyspark/sql/types.py

```diff
@@ -244,10 +244,39 @@ class FractionalType(NumericType):
     """Fractional data types."""
 
 
-class StringType(AtomicType, metaclass=DataTypeSingleton):
-    """String data type."""
+class StringType(AtomicType):
+    """String data type.
 
-    pass
+    Parameters
+    ----------
+    collationId : int
+        the collation id number.
+    """
+
+    collationNames = [
+        "UCS_BASIC",
+        "UCS_BASIC_LCASE",
+        "UNICODE",
+        "UNICODE_CI"
+    ]
+
+    def __init__(self, collationId: int = 0):
+        self.collationId = collationId
+
+    def collationIdToName(self, collationId: int) -> str:
+        return "(" + StringType.collationNames[collationId] + ")" if collationId != 0 else ""
+
+    def collationNameToId(collationName: str) -> int:
+        return StringType.collationNames.index(collationName) if collationName != "" else 0
+
+    def simpleString(self) -> str:
+        return "string%s" % (self.collationIdToName(self.collationId))
+
+    def jsonValue(self) -> str:
+        return "string%s" % (self.collationIdToName(self.collationId))
```

Comment:
Is this matched with `DataType.COLLATED_STRING_TYPE` in Scala? It has to be same and able to parse via `DataType.parseDataType`

Reply:
@HyukjinKwon Aligned this with collated string regex from Scala and added more tests to confirm parsing works as you suggested, please check the latest changes.